File: Java/DiamondPattern.java
import java.util.Scanner;
public class DiamondPattern {
	public static void main(String[] args) {
		
		System.out.println("Enter a number for rows: ");
		Scanner scan = new Scanner(System.in);
		
	    int rows = scan.nextInt();
		
		System.out.println("Enter the symbol for the pattern: ");
		String s = scan.next();
		
		int columns = rows - 1;
		int i, j = 1;
		for (j = 1; j<= rows; j++) {
			for (i = 1; i<= columns; i++) {
				System.out.print(" ");
			}
				
			columns--;
			for (i = 1; i <= 2 * j - 1; i++) {  
				System.out.print(s);
			}
			 
			System.out.println("");
		}
				  
		columns = 1;  
		for (j = 1; j<= rows - 1; j++) {  
			for (i = 1; i<= columns; i++){  
				System.out.print(" ");  
			}  
			columns++;  
			for (i = 1; i<= 2 * (rows - j) - 1; i++){  
				System.out.print(s);  
			}  
			System.out.println("");
		}
	}
}
		

File: C/diamond_pattern.c
// GitHub user --- aditya-464
// Problem - Print the Diamond pattern given below. (Number of rows are odd!)
//    *
//   ***
//  *****
// *******
//  *****
//   ***
//    *


#include <stdio.h>
int main()
{
    int rows;
    printf("The number of rows should be ODD !\n");
    printf("Enter the number of rows : ");
    scanf("%d", &rows);
    int i, j, k, upRows;
    upRows = rows / 2;

    // For upper part
    int limit = 1;
    for (i = 0; i <= upRows; i++)
    {
        for (j = 0; j < upRows - i; j++)
        {
            printf(" ");
        }
        for (k = 0; k < limit; k++)
        {
            printf("*");
        }
        printf("\n");
        limit += 2;
    }
    limit -= 4;

    // For lower part
    for (i = upRows - 1; i >= 0; i--)
    {
        for (j = 0; j < upRows - i; j++)
        {
            printf(" ");
        }
        for (k = 0; k < limit; k++)
        {
            printf("*");
        }
        printf("\n");
        limit -= 2;
    }

    return 0;
}